core/templates/dev/head/base.js

```diff
@@ -95,6 +97,11 @@ oppia.controller('Base', [
 
     // If this is nonempty, the whole page goes into 'Loading...' mode.
     $rootScope.loadingMessage = '';
+    // Method to call windowSize on resizing the window
+    $window.onresize = function() {
```

Comment:
I disagree, because it's somewhat implicit -- there's no canonical registry of stuff bound to $rootScope. I prefer using the service because then at least the dependency is explicitly specified.

To clarify the second part a bit more, I meant something like, in windowDimensionsService, adding a private variable to store on-resize hook functions:

    var onResizeHooks = [];

and a public API method:

    addOnResizeHook: function(hookFn) {
      onResizeHooks.push(...);
    };

and adding a global listener:

    $window.onresize = function() {
      onResizeHooks.forEach(function(hookFn) {
        hookFn();
      };
    };

Does this sound sensible?